db/view/view.cc

```diff
@@ -2269,7 +2269,7 @@ future<> view_builder::calculate_shard_build_step(view_builder_init_state& vbi)
     }
 
     for (auto& [_, build_step] : _base_to_build_step) {
-        boost::sort(build_step.build_status, [] (view_build_status s1, view_build_status s2) {
+        std::ranges::sort(build_step.build_status, [] (view_build_status s1, view_build_status s2) {
             return *s1.next_token < *s2.next_token;
```

Comment:
projection (but careful to retain the indirection)

Reply:
Does it worth using projection then?
It seems more straightforward to keep using the comparator function method.